### PR TITLE
[ODS-6159] Migrate Ed-Fi-Ods solution to .NET 8 

### PR DESCRIPTION
**Pull request by simpat-jesus**

### CODE REVIEW

File: Application/EdFi.Ods.Api/Helpers/TypeHelper.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ private static List<(Assembly assembly, Type type)> GetImplementationsOf<T>()
         {
             var allModules = AppDomain.CurrentDomain.GetAssemblies()
                 .SelectMany(
-                    a => a.GetTypes()
+                    a => a.GetExportedTypes()
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/Helpers/TypeHelper.cs:44]: workaround for [compatibility issue](https://github.com/dotnet/SqlClient/issues/2228)

File: Application/EdFi.Ods.Features/ChangeQueries/Controllers/DeletesController.cs [hunk 2]
```diff
@@ -106,7 +106,7 @@ public async Task<IActionResult> Get(string schema, string resource, [FromQuery]
             // Add the total count, if requested
             if (urlQueryParametersRequest.TotalCount)
             {
-                Response.Headers.Add("Total-Count", deletedItemsResponse.Count.ToString());
+                Response.Headers.Append("Total-Count", deletedItemsResponse.Count.ToString());
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Features/ChangeQueries/Controllers/DeletesController.cs:109]: Fixes warning [ASP0018](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0019?view=aspnetcore-8.0)

File: Application/EdFi.Ods.Features/ChangeQueries/Controllers/KeyChangesController.cs [hunk 3]
```diff
@@ -109,7 +109,7 @@ public class KeyChangesController : ControllerBase
             // Add the total count, if requested
             if (urlQueryParametersRequest.TotalCount)
             {
-                Response.Headers.Add("Total-Count", keyChangesResponse.Count.ToString());
+                Response.Headers.Append("Total-Count", keyChangesResponse.Count.ToString());
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Features/ChangeQueries/Controllers/KeyChangesController.cs:112]: Fixes warning [ASP0018](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0019?view=aspnetcore-8.0)

File: Application/EdFi.Ods.Features/ChangeQueries/ActionResults/SnapshotsAreReadOnlyResult.cs [hunk 4]
```diff
@@ -37,7 +38,7 @@ public async Task ExecuteResultAsync(ActionContext context)
                 StatusCode = (int) HttpStatusCode.MethodNotAllowed,
             };
 
-            context.HttpContext.Response.Headers.Add("Allow", "GET");
+            context.HttpContext.Response.Headers.Append("Allow", "GET");
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Features/ChangeQueries/ActionResults/SnapshotsAreReadOnlyResult.cs:41]: Fixes warning [ASP0018](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0019?view=aspnetcore-8.0)

File: Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs [hunk 5]
```diff
@@ -215,7 +215,7 @@ private string GetReadContentType()
             // Return multiple results
             if (queryParameters.TotalCount)
             {
-                Response.Headers.Add(HeaderConstants.TotalCount, result.ResultMetadata.TotalCount.ToString());
+                Response.Headers.Append(HeaderConstants.TotalCount, result.ResultMetadata.TotalCount.ToString());
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/Controllers/DataManagementControllerBase.cs:218]: Fixes warning [ASP0018](https://learn.microsoft.com/en-us/aspnet/core/diagnostics/asp0019?view=aspnetcore-8.0)

File: Application/EdFi.Ods.Api/Middleware/EdFiOAuthAuthenticationHandler.cs [hunk 6]
```diff
@@ -33,9 +33,8 @@ public class EdFiOAuthAuthenticationHandler : AuthenticationHandler<Authenticati
             IOptionsMonitor<AuthenticationSchemeOptions> options,
             ILoggerFactory loggerFactory,
             UrlEncoder encoder,
-            ISystemClock clock,
             IOAuthTokenAuthenticator oauthTokenAuthenticator)
-            : base(options, loggerFactory, encoder, clock)
+            : base(options, loggerFactory, encoder)
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/Middleware/EdFiOAuthAuthenticationHandler.cs:37]: Fix obsolete warnings related to [ISystemClock](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/8.0/isystemclock-obsolete)

File: Application/EdFi.Ods.Api/OperationalContext/PersonIdentification/AmbiguousUniqueIdMatchException.cs [hunk 7]
```diff
@@ -28,10 +28,5 @@ public AmbiguousUniqueIdMatchException(string message)
 
         public AmbiguousUniqueIdMatchException(string message, Exception inner)
             : base(message, inner) { }
-
-        protected AmbiguousUniqueIdMatchException(
-            SerializationInfo info,
-            StreamingContext context)
-            : base(info, context) { }
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/OperationalContext/PersonIdentification/AmbiguousUniqueIdMatchException.cs:35]: Fix warning related to [legacy serialization](https://github.com/dotnet/docs/issues/34893)

File: Application/EdFi.Ods.Api/OperationalContext/PersonIdentification/NoUniqueIdMatchException.cs [hunk 8]
```diff
@@ -28,10 +28,5 @@ public NoUniqueIdMatchException(string message)
 
         public NoUniqueIdMatchException(string message, Exception inner)
             : base(message, inner) { }
-
-        protected NoUniqueIdMatchException(
-            SerializationInfo info,
-            StreamingContext context)
-            : base(info, context) { }
```
Comment (simpat-jesus) [inline Application/EdFi.Ods.Api/OperationalContext/PersonIdentification/NoUniqueIdMatchException.cs:35]: Fix warning related to [legacy serialization](https://github.com/dotnet/docs/issues/34893)

**Review by semalaiappan — APPROVED** [review]

**Pull request merged**